src/librustc_session/lint/builtin.rs

```diff
@@ -46,6 +46,18 @@ declare_lint! {
     "shift exceeds the type's number of bits"
 }
 
+declare_lint! {
+    pub OVERFLOW,
```

Comment:
My inclination would be to merge this with `EXCEEDING_BITSHIFTS` into one lint, but I am not sure what the procedure is for changing/removing existing lint names.

Also, the new lint names are maybe not great, suggestions are welcome :)

Reply:
Well, having a separate lint for bitshifts seems silly. I mean, if that's the way to go we should have separate lints for negation, addition, subtraction, ...

Reply:
I agree, let's do it in a separate PR though and ping some ppl on it

Reply:
Seems like a design question that ought to be bumped to the language team?

Reply:
Is the lang team involved with the details of lint naming? I would have said this is compiler team territory but what do I know.^^

Reply:
> How about overflowing_expressions

An alternative is `arithmetic_overflow`.

Reply:
OTOH, if we want to extend this to include shifts later... do they count as "arithmetic"?

Reply:
The lang team probably doesn't care about naming per se, but granularity and lint levels, probably yes.

Reply:
Although if we can reach a reasonable conclusion here that has some consensus re. grouping then I think we could also keep things here and not "escalate".

Reply:
Well, one thing we have to do in this PR is pick a name for the new lint. Proposals so far, and some new ones:
* overflow
* overflowing_expressions
* arithmetic_overflow
* integer_overflow

Whether or not this also covers overflowing bitshifts is orthogonal insofar as that doesn't impact the choice of name.

(Also I made the choice of making this lint separate from the one that warns about panics. They were previously all together, but the thing is that overflow doesn't always panic, so it seems to make sense to separate that.)